### PR TITLE
Alibi detect runtime updates

### LINKED ISSUE #494: Remove support for legacy protocols

### PR DESCRIPTION
**Pull request by SachinVarghese**
This PR updates the alibi-detect runtime in MLserver project. This PR does the following things, 
- Removes the support for legacy protocols and updates the docs (https://github.com/SeldonIO/MLServer/issues/494)
- Updates runtime to use only the load detector fuctionality and removes the ability to initialise the detector based on model atrifacts and other configuration.

This is a step towards https://github.com/SeldonIO/MLServer/issues/496 to fully support the new config based detectors subject to this alibi-detect PR https://github.com/SeldonIO/alibi-detect/pull/389 merging. Either way current code changes supports loading detectors based on configs or even dill files directly.

Fixes #494 #496

### CODE REVIEW

**Comment by SachinVarghese** [comment]
New updated docs https://github.com/SachinVarghese/MLServer/blob/alibi-detect/docs/examples/alibi-detect/README.ipynb

**Comment by SachinVarghese** [comment]
Action Items:
- Simplify examples (remove extra drift and outlier stuff)
- Add docs links
- Predict kwargs move to model settings

File: docs/examples/alibi-detect/README.md [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
-## Reference Data and Configuration
+## Fetch reference data
 
-The first step will be to fetch a reference data for an `alibi-detect` model. For that, we will use the [income Classifier example from the `alibi-detect` documentation](https://docs.seldon.io/projects/alibi-detect/en/latest/examples/cd_chi2ks_adult.html)
+The first step will be to fetch a reference data and other relevant metadata for an `alibi-detect` model. 
 
-Install Alibi dependencies for dataset and detector creation
+For that, we will use the [alibi](https://github.com/SeldonIO/alibi) library to get the adult dataset with [demographic features from a 1996 US census](https://archive.ics.uci.edu/ml/datasets/census+income).
+
+**Note**: Install `alibi` dataset dependencies and `alibi-detect` for detector configuration
```
Comment (adriangonz) [inline docs/examples/alibi-detect/README.md:11]: Not a big deal, but we could use Myst annotations on this one, e.g.
````
```{note}
Install `alibi` dataset dependencies and `alibi-detect` for detector configuration
```
````

, which would render a block like the one [here](https://mlserver.readthedocs.io/en/latest/examples/custom/README.html#deploying-our-custom-image)

File: docs/examples/alibi-detect/README.md [hunk 2]
```diff
@@ -75,35 +106,35 @@ For that, we will need to create 2 configuration files:
 }
 ```
 
+    Overwriting settings.json
```
Comment (adriangonz) [inline docs/examples/alibi-detect/README.md:109]: Should we clear the outputs from the notebook? Again, not a big deal (i.e. feel free to skip this one).

File: runtimes/alibi-detect/tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+TESTS_PATH = os.path.dirname(__file__)
+TESTDATA_PATH = os.path.join(TESTS_PATH, "testdata")
+
+
+def pytest_collection_modifyitems(items):
+    """
+    Add pytest.mark.asyncio marker to every test.
+    """
+    for item in items:
+        item.add_marker("asyncio")
```
Comment (adriangonz) [inline runtimes/alibi-detect/tests/conftest.py:15]: If you rebase from `master` (this PR in particular: #507 ), this bit shouldn't be required anymore.

**Review by adriangonz — APPROVED** [review]
This one looks great @SachinVarghese! I love how simple the runtime is now. The tests also look great, they'll be massively helpful to catch any unwanted change (either in code or deps). :rocket: 

I've added a small couple comments, they're mainly nitpicks though (i.e. feel free to either address them or leave it as it is). 

Once the tests look green, we should be good to go ahead with this one :+1:

File: runtimes/alibi-detect/tests/conftest.py (hunk 3 above)
Comment (SachinVarghese) [reply runtimes/alibi-detect/tests/conftest.py]: Fixed now in 8cdec90

File: docs/examples/alibi-detect/README.md (hunk 2 above)
Comment (SachinVarghese) [reply docs/examples/alibi-detect/README.md:101]: Fixed now in 8cdec90

File: docs/examples/alibi-detect/README.md (hunk 1 above)
Comment (SachinVarghese) [reply docs/examples/alibi-detect/README.md]: Fixed now in 8cdec90

**Pull request merged**